go/proof.go

```diff
@@ -92,18 +93,11 @@ func (p *CommitmentProof) Calculate() (CommitmentRoot, error) {
 
 // Verify does all checks to ensure this proof proves this key, value -> root
 // and matches the spec.
-func (p *ExistenceProof) Verify(spec *ProofSpec, root CommitmentRoot, key []byte, value []byte) error {
+func (p *ExistenceProof) Verify(spec *ProofSpec, root CommitmentRoot) error {
```

Comment:
Would it be possible to make this private? I think it'd be good if ics23 only exposed `VerifyMembership` and `VerifyNonMembership` as it's two public APIs